scio-core/src/main/scala/com/spotify/scio/values/PairHashSCollectionFunctions.scala

```diff
@@ -79,7 +83,7 @@ class PairHashSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
    */
   def hashJoin[W: Coder](
     sideInput: SideInput[Map[K, Iterable[W]]]
-  )(implicit koder: Coder[K], voder: Coder[V]): SCollection[(K, (V, W))] =
+  ): SCollection[(K, (V, W))] =
```

Comment:
We can technically remove these `: Coder` too and use `sideInput.view.getPCollection.getCoder`. But we'll have to extract `Coder[W]` from the side input's `Coder[Map[K, Iterable[W]]`. Do we want to go that far?

@regadas opinion?

Reply:
I think that since we are cleaning all of these up we might as well do it everywhere we can if does do add extra complexity

Reply:
Added a helper to extract coders from `SideInput[Map[K, Iterable[V]]`. That seems to be the only case we need to extract coders from a side input. There're 2 possibilities for this case though, one a Scala `Coder[Map[K, Iterable[V]]]` for `asMapSingletonSideInput` and one Beam `KvCoder[VoidCoder, KvCoder[K, V]]` for `asMultiMapSideInput`, where the coder is inherited from the underlying `PCollection[KV[K, V]]`